### ISSUE TITLE
Verification

### ISSUE THREAD

**Issue by Adam-Burke** [stormpath/express-stormpath#583]
Sorry, I'm hoping someone can point me in the right direction.

I want to verify a users email and then redirect to my app. 

Is it possible to verify with stormpath directly then redirect to my login page? The Workflows and emails page in the stormpath ui only has Link Base URL which I guess means I have to provide the route for verification myself?

How do I do this using the JSON API stateless version of the server. Do I need to use the nodejs sdk?
Is the client object and tenant object from that sdk available in express-stormpath. It seems the verifyEmail function is only available on the tenant object. 

I think I'm missing something. 

Cheers,

Adam

**Comment by mdeggies** [comment]
Hey @Adam-Burke! If you're using express-stormpath, all you need to do is set the link base URL to ``http://localhost:3000/verify``, and then [enable that route](https://github.com/stormpath/express-stormpath/blob/master/lib/config.yml#L122) in the Stormpath config section in your server file: 

````
app.use(stormpath.init(app, {
  ....
  web: {
    verifyEmail: {
      enabled: true,
      nextUri: "/login"
    }
  }
});
````

For future reference, you can configure a whole mess of things in app.use(stormpath.init...). This is a good place to start: https://docs.stormpath.com/nodejs/express/latest/configuration.html. It also links to 'Web Configuration Defaults' which contains a full list of the things you can customize: https://github.com/stormpath/express-stormpath/blob/master/lib/config.yml. 

LMK if this helps or if you have any follow-up q's!

**Comment by Adam-Burke** [comment]
Yeah I placed that config in but it is just is a missing route on the server. Doesn't that only work if you are using express-stormpath as traditional web app or spa? I'm only using it for auth and to dispense json.

The localhost:3001 server is just my localhost dev ur where my app is being servedl. I'm serving the html seperately from the express-stormpath server. They're on different ports but cors is enabled. 

    expand: {
      customData: true,
      groups: true,
      groupMemberships: true,
      providerData: true
    },
    web: {
     produces: ['application/json'],
     verifyEmail: {
        enabled: true,
        nextUri: 'http://localhost:3001/login'
     }
    }

**Comment by Tweety-FER** [comment]
@Adam-Burke One thing to note is that the email verification workflow also requires the whole thing to be enabled in the Directory used by your application. You can do this in the dashboard in api.stormpath, under Directory/Workflows & Emails. Just ran into this myself and figured it might help.
